### PR TITLE
Allow configuring repo in different scope (for example, if user scope…

### PR DESCRIPTION
**Pull request by eflumerf**
… is disabled). Also, allow specifying MPD user directory.

Tested using `spack -d mpd init -u ./.mpd -r site -f`

### CODE REVIEW

File: mpd/config.py [hunk 1]
```diff
@@ -46,6 +47,10 @@ def selected_project_token():
 
 
 def user_config_dir():
+    dir_from_config = spack.config.get('config:mpd_user_dir')
+    tty.debug(f"Directory from config is {dir_from_config}")
+    if dir_from_config is not None and Path(dir_from_config).exists():
+      return Path(dir_from_config).resolve()
     return (Path.home() / ".mpd").resolve()
```
Comment (knoepfel) [inline mpd/config.py:54]: I'm wondering if we can change all this to just:

```suggestion
    return spack.config.get('config:mpd_user_dir')
```

and then do...

File: mpd/init.py [hunk 2]
```diff
@@ -28,16 +29,26 @@ def setup_subparser(subparsers):
         action="store_true",
         help='assume "yes" is the answer to confirmation request for reinitialization',
     )
+    init.add_argument("-u", "--user-dir", action="store", help="Set MPD configuration directory (default ~/.mpd)")
```
Comment (knoepfel) [inline mpd/init.py:32]:
```suggestion
    init.add_argument("-u", "--user-dir", 
                      action="store", 
                      help="Set MPD configuration directory (default ~/.mpd)",
                      default=(Path.home() / ".mpd").resolve())
```

File: mpd/init.py [hunk 3]
```diff
@@ -28,16 +29,26 @@ def setup_subparser(subparsers):
...
 
 def process(args):
     spack_root = spack.paths.prefix
-    local_dir = config.user_config_dir()
+
+    if args.user_dir is not None:
+        local_dir = Path(args.user_dir).resolve()
+        spack.config.set('config:mpd_user_dir', str(local_dir))
+    else:
+        local_dir = config.user_config_dir()
```
Comment (knoepfel) [inline mpd/init.py:50]: This all becomes:

```suggestion
    local_dir = Path(args.user_dir).resolve()
    spack.config.set('config:mpd_user_dir', str(local_dir))
```

**Review by knoepfel — COMMENTED** [review]
@eflumerf, the changes look good in general.  I added a few suggestions.  See what you think.

**Comment by eflumerf** [comment]
@knoepfel Committed, thanks!

**Comment by knoepfel** [comment]
An excellent improvement, Eric.  Thanks.

**Pull request merged**